echidna/core/smear.py

```diff
@@ -389,3 +389,32 @@ def weight_gaussian_radius_spectra(self, true_spectrum, num_sigma=5.):
                         i += 1
         smeared_spectrum._raw_events = raw_events
         return smeared_spectrum
+
+
+class EResSmear(Smear):
+    """ Allows you to smear directly by energy resolution (e.g. 0.05 * E)
+
+    Inherits from :class:`Smear` to provide exactly the same functionality
+    but overrides :meth:`Smear.get_energy_sigma` to use an energy 
+    resolution to calculate sigma.
+
+    Attributes:
+      _light_yield (float): Number of PMT hits expected for a
+        MeV energy deposit in NHit/MeV
+      _energy_resolution (float): energy resolution e.g. 0.05 for 5%. 
+      _position_resolution (float): Sigma in mm
+    """
+    def __init__(self, energy_resolution):
+        super(EResSmear, self).__init__()
+        self._energy_resolution = energy_resolution
+
+    def get_energy_sigma(self, energy):
+        """ Calculates sigma at a given energy.
+
+        Args:
+          energy (float): Energy value of data point(s)
+
+        Returns:
+          float. Sigma equivalent to energy * energy_resolution
+        """
+        return energy * self._energy_resolution
```

Comment:
The reason I added this was for when you want to smear truth `Spectra` based on the energy resolution for another experiment, i.e. KamLAND-Zen. Our light yield values (in NHit per MeV) are fairly specific to SNO+ and although other experiments may uses it (or a similar parameter), it doesn't seem to be something they publish.

What they do publish though is just the energy resolution expressed as a percentage e.g. 6.6% / sqrt(E[MeV]). I added this method to deal with cases where this is the only information you have.

And then the reason why it is a new (daughter) class, is so that it has all the functionality of the original class, but just overrides the original method for calculating sigma.